qp_klp/tests/test_step.py

```diff
@@ -602,3 +682,143 @@ def test_get_tube_ids_from_qiita(self):
                          '12.15': '12.15'}}
 
         self.assertDictEqual(obs, exp)
+
+    def test_compare_samples_against_qiita(self):
+        fake_client = FakeClient()
+        step = Step(self.pipeline, self.qiita_id, None)
+        results = step._compare_samples_against_qiita(fake_client)
+
+        # confirm projects in results match what's expected
+        obs = set([x['project_name'] for x in results])
```

Comment:
why do you have to reiterate and convert to a set to test? Shouldn't you be testing the output of the function without any transformations?

Reply:
That's only true because the call to Qiita is faked and the order is fixed, but I get you.